src/sentry/eventstore/snuba/backend.py

```diff
@@ -32,10 +32,9 @@ def get_events(
         Get events from Snuba.
         """
         assert filter, "You must provide a filter"
-
         cols = self.__get_columns(additional_columns)
 
-        result = snuba.raw_query(
+        result = snuba.dataset_query(
```

Comment:
There is a risk of making the wrong dataset decision here if a user has got a tag that is a transaction column

Reply:
Good point. I'll coerce the dataset here.

Reply:
I think we also need to do this everywhere we call `eventstore.get_events`. Similar to https://github.com/getsentry/sentry/pull/15254

Reply:
We can't do it everywhere as there are a handful of endpoints that need to access both types of events.

We could add `dataset` onto `eventstore.Filter` as you/manu had in another pull request. This would let us force the correct dataset in other error only endpoints.

Reply:
We can get around this by just including the condition `["type", "!=", "transaction"]` anywhere this gets called where we want to force the right dataset.